tests/functional/oauth_sign_up_verification_redirect.js

```diff
@@ -74,28 +79,48 @@ define([
       function countSignInAndInvalidState() {
         return function () {
           return this.parent
-            .findByCssSelector('body')
-              .getVisibleText()
-              .then(function (text) {
-                var isBadRequest = text === 'Bad request - missing code - missing state';
-                if (isBadRequest) {
-                  badRequestCount++;
-                }
-              })
-            .end()
-
-            .setFindTimeout(0)
-            .findByCssSelector('#loggedin')
-              .then(function () {
-                signedInCount++;
-              }, function (err) {
-                if (/NoSuchElement/.test(String(err))) {
-                  // swallow the error
-                  return;
-                }
-
-                throw err;
-              })
+            // This test can do redirecting. We want to make sure all redirections
+            // have completed. The user will either finish at 123done's logged in
+            // screen, or 123done's /api/oauth endpoint where an error message will
+            // be displayed.
+            // As ugly as it is, wait a few seconds for the redirects to finish.
+            // Check which URL the user is at. If they are at the /api/oauth endpoint,
+            // check for the error message. If they are at 123done, then check for
+            // the #loggedin element.
+
+            .sleep(5000) // <--- GROSS!!! Read above.
```

Comment:
I went down that path and hit some strangeness that I wasn't sure how to overcome. The user *always* hits the /api/auth endpoint, and sometimes stops there, sometimes continues to 123done. Even a poll would have to timeout eventually, if it's expecting to land eventually on 123done.

Now that I think about it. It could go something like this:

```js
.then(pollFor123doneUrl()
.then(function () {
  // user is at 123done
  successCount++
}, function (err) {
  if (err == POLL_TIMEOUT) {
   failureCount++
  }
  throw err
})
```